### PR TITLE
Add a test recipe for testing the speed command

### PR DESCRIPTION
**Pull request by mattcaswell**
Previously there was no test for the speed command. We just do some simple
testing, running the command with various options to confirm that it doesn't
crash or report errors. We use the new -testmode option to ensure that this
happens quickly and doesn't really run full speed tests.

### CODE REVIEW

**Comment by mattcaswell** [comment]
I'm hoping this will pull our coveralls % up a bit since there was no coverage at all for the apps/speed.c file.

File: test/recipes/20-test_speed.t [hunk 1]
```diff
@@ -0,0 +1,119 @@
+#! /usr/bin/env perl
+# Copyright 2017-2023 The OpenSSL Project Authors. All Rights Reserved.
```
Comment (nhorman) [inline test/recipes/20-test_speed.t:2]: nit: copyright date

**Review by nhorman — APPROVED** [review]
just the copyright nit, otherwise looks good

**Review by Sashan — APPROVED** [review]
looks good to me as far as I can tell.

**Review by paulidale — APPROVED** [review]

**Comment by paulidale** [comment]
With the copyright date fix.

**Comment by mattcaswell** [comment]
Fixed the copyright date issue. Also fixed a memory leak in speed that the test revealed. Also changed speed to not compile in support for DSA if DSA support has also been compiled out of the library (in the same way as we do for lots of other algorithms). Finally numerous fixups in the test itself to skip testing of various options where support for that option is not available for some reason.

Please take another look.

**Comment by mattcaswell** [comment]
CIs are still running but looking promising...

**Review by tom-cosgrove-arm — APPROVED** [review]
LGTM

**Review by nhorman — APPROVED** [review]

**Review by t8m — APPROVED** [review]

**Comment by t8m** [comment]
Merged to the master branch. Thank you.

**Pull request closed without merging**